src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -27,11 +32,18 @@
 from pants.option.subsystem import Subsystem
 from pants.util.docutil import doc_url, git_url
 from pants.util.meta import classproperty
+from pants.util.pip_requirement import PipRequirement
 from pants.util.strutil import softwrap
 
 logger = logging.getLogger(__name__)
 
 
+@dataclass(frozen=True)
+class PackageNameAndVersion:
```

Comment:
Potentially these new types and methods could be private, so that external plugins don't rely on them (and thus force us to keep them more stable)?

```suggestion
class _PackageNameAndVersion:
```

and `_install_from_resolve_help` and `_default_package_name_and_version`.

What do you think?

Reply:
Done - made all of these private